### ISSUE TITLE
qt5agg image quality changes when window is out of focus

### ISSUE THREAD

**Issue by dcherian** [matplotlib/matplotlib#13012]
### Bug report

**Bug summary**

Quality of qt rendered figures is drastically reduced when window is moved out of focus and then back in focus.

**Code for reproduction**

```python
import matplotlib.pypot as plt
plt.plot([1, 2, 3, 4], [5, 5.5, 6.5, 8.5], alpha=0.5)
```

**Actual outcome**

![mpl-before](https://user-images.githubusercontent.com/2448579/50118171-c4b71100-0203-11e9-9bbf-1cc763aa1525.png)

Switch to different window and back:
![mpl-after](https://user-images.githubusercontent.com/2448579/50118177-c7b20180-0203-11e9-9e72-f26dbbfed192.png)

At this stage, calling `plt.draw()` fixes it till the window goes out of focus. Zooming in on the figure also returns it to the earlier quality. Note that transparency is lost.

**Expected outcome**

Rendering doesn't change.

**Matplotlib version**

  * Operating system: Ubuntu 16.04
  * Matplotlib version: 3.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.7
  * Jupyter version (if applicable): console 6.0.0
  * Other libraries: pyqt  5.6.0            py36h8210e8a_7    conda-forge

everything installed from conda-forge using conda.

**Comment by WeatherGod** [comment]
Does this happen with any other backend? I wonder if this might be
something Ubuntu's window manager might be doing. You wouldn't happen to be
using something custom like Mint or Cinnamon? Which desktop manager are you
using?

**Comment by dcherian** [comment]
I'm using Xfce and the Tk backend works fine.

**Comment by tacaswell** [comment]
> Note that transparency is lost.

Do you have other transparency than the alpha on the line?

**Comment by dcherian** [comment]
No, but `axes.facecolor=None` and `figure.facecolor=None`. Does that count?

**Comment by tacaswell** [comment]
Yes, that is likely the critical detail!  If you repeatedly focus / defocus does it get worse?

I think what is going on is that for GUI backends we use Agg to render and RGBA buffer and then use the GUI frameworks `drawImage` or equivalent to paint the figure onto your screen (see https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_qt5agg.py#L23) for details.  If you set the Figure and Axes color to `None` the alpha on most of the rendered figure is 0.  Unfortunately, we only clear the Qt painter right after triggering a draw (https://github.com/matplotlib/matplotlib/blob/529410fa0c9379cd60b1e4d8fdd0478bb2f7e892/lib/matplotlib/backends/backend_qt5agg.py#L41-L43 ).  On a repaint where we did re-draw we just re-use the same buffer and draw to the screen.  If you have an opaque background, this is OK as when Qt does the compositing the opaque layer is the only thing that survives, however all of your pixels have non-zero alpha so we are repeatedly compositing it with its self.

If you delete the 3 lines above does the problem go away?

**Comment by dcherian** [comment]
> If you repeatedly focus / defocus does it get worse?
Yes! 

When I remove those three lines my figure background is black.
